theme/cdnjs_hashes.j2

```diff
@@ -1,6 +1,21 @@
 {%- set cdnjs_hashes = {
+  "font-awesome": {
+    "version": "4.7.0",
+    "assets": {
+      "css/font-awesome.css": "sha384-FckWOBo7yuyMS7In0aXZ0aoVvnInlnFMwCv77x9sZpFgOonQgnBj1uLwenWVtsEj",
+      "css/font-awesome.css.map": "sha384-wWYGaw6eOdtlU+9Qvm8knJM1z+Eu6gb3E15DfXAAUjYtuR1aAjl9AS231tqdLKlo",
+      "css/font-awesome.min.css": "sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN",
+      "fonts/FontAwesome.otf": "sha384-7AxH7c5DMKm53fxbc2HaPopOPZBE7W43HdFmP2Itt4aQS9DDq9rkE0IzE2ldLvmw",
+      "fonts/fontawesome-webfont.eot": "sha384-aT4Js4F9nY6/YfdaUe74bgxzlDR1S188jXIJZQj981StDN6YfBNOetz0JVYt8RRs",
+      "fonts/fontawesome-webfont.svg": "sha384-cpmd2Pb8rcQw0Rb0JqVxqpRf1aPgEYmq/SvC4aU562SjUEQ+KClejtxnrYLwTOI1",
+      "fonts/fontawesome-webfont.ttf": "sha384-arWNUkFcwgK/p/bP6IcquY+ZOXqnmOVkdFBJpkkKX7ldohEAUno5hxI+lm4tn4Cm",
+      "fonts/fontawesome-webfont.woff": "sha384-miAIzLueFC2a1i22Q520oFQwc/N2wtaYe93QyrKYlj5f0T0+DkQEpMs7S+MvkTW/",
+      "fonts/fontawesome-webfont.woff2": "sha384-wmfasCsFE3p9BgQSCciiJ0R6GvOSsKKZJMXBvTiCuWa3ypa5yXGVjSdzuU/ON6P6"
```

Comment:
I like how you moved the version number to the json, rather than having to mention it in the code.
Also, getting rid of binaries in the source is always great :)
Where are you getting these hashes from, though?

Reply:
I wrote a script that interrogates the API for CDNJS (to get a list of all assets for a given library + version), then fetches them, hashes them, and records the hash in `cdnjs_hashes.j2 `.

It's all submitted upstream as https://github.com/mkdocs/mkdocs/pull/1906

Reply:
awesome :+1:

Reply:
hmm, perhaps it could be useful to include the script here, and wrap it in a makefile command to update the hashes based on the resources we want to use? basically whenever we want to use a new version of something, we would
```sh
make update-hashes RESOURCE=package:version
```

Reply:
It's in that linked pull request: https://github.com/mkdocs/mkdocs/pull/1906/files#diff-45ccb2a02cb25e2842856224f8e63880

The process for changing a package version is just:
- update the version in `cdnjs_libraries.json`;
- rerun the script.

Reply:
> getting rid of binaries in the source is always great :)

I'm actually tempted to do some git surgery to excise those binaries from the history entirely, since otherwise people will still download them while cloning the repository.  :smiling_imp: 
The entanglement with #8 makes that a bit iffy though  :3

Reply:
You can always do it later.
An alternative is to point out in the readme that users can [`git clone --depth n`](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt) to avoid cloning too much

Or delete the entire repository history before anyone else starts working on it :D gotta love `git checkout master; git reset <initial commit>; git commit -m 'sup'; git push --force` <3

Reply:
> You can always do it later.
Rewriting the history after it's been merged to `main` will cause conflicts for everyone, though.

Reply:
Done